src/pkg/utils/checksum.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// ValidatePackageChecksums validates the checksums of a Zarf package.
+func ValidatePackageChecksums(baseDir string, pathsToCheck []string) error {
+	spinner := message.NewProgressSpinner("Validating package checksums")
+	defer spinner.Stop()
+
+	// Run pre-checks to make sure we have what we need to validate the checksums
+	if InvalidPath(baseDir) {
+		return fmt.Errorf("invalid base directory: %s", baseDir)
+	}
+	var pkg types.ZarfPackage
+	err := ReadYaml(filepath.Join(baseDir, config.ZarfYAML), &pkg)
+	if err != nil {
+		return err
+	}
+	aggregateChecksum := pkg.Metadata.AggregateChecksum
+	if aggregateChecksum == "" {
+		return fmt.Errorf("missing aggregate checksum")
+	}
+	if len(aggregateChecksum) != 64 {
+		return fmt.Errorf("invalid aggregate checksum: %s", aggregateChecksum)
+	}
+	isPartial := false
+	if len(pathsToCheck) > 0 {
+		pathsToCheck = Unique(pathsToCheck)
+		isPartial = true
+		message.Debugf("Validating checksums for a subset of files in the package - %v", pathsToCheck)
+		for idx, path := range pathsToCheck {
+			pathsToCheck[idx] = filepath.Join(baseDir, path)
+		}
+	}
+
+	pathCheckMap, err := PathCheckMap(baseDir)
+	if err != nil {
+		return err
+	}
+
+	checksumPath := filepath.Join(baseDir, config.ZarfChecksumsTxt)
+	actualAggregateChecksum, err := GetSHA256OfFile(checksumPath)
+	if err != nil {
+		return fmt.Errorf("unable to get checksum of: %s", err.Error())
+	}
+	if actualAggregateChecksum != aggregateChecksum {
+		return fmt.Errorf("invalid aggregate checksum: (expected: %s, received: %s)", aggregateChecksum, actualAggregateChecksum)
+	}
+
+	pathCheckMap[filepath.Join(baseDir, config.ZarfChecksumsTxt)] = true
+	pathCheckMap[filepath.Join(baseDir, config.ZarfYAML)] = true
+	pathCheckMap[filepath.Join(baseDir, config.ZarfYAMLSignature)] = true
+
+	err = LineByLine(checksumPath, func(line string) error {
+		split := strings.Split(line, " ")
+		sha := split[0]
+		rel := split[1]
+		if sha == "" || rel == "" {
+			return fmt.Errorf("invalid checksum line: %s", line)
+		}
+		path := filepath.Join(baseDir, rel)
+
+		status := fmt.Sprintf("Validating checksum of %s", rel)
+		spinner.Updatef(message.Truncate(status, message.TermWidth, false))
+
+		if InvalidPath(path) {
+			if !isPartial && !pathCheckMap[path] {
+				return fmt.Errorf("unable to validate checksums - missing file: %s", rel)
+			} else if SliceContains(pathsToCheck, path) {
+				return fmt.Errorf("unable to validate partial checksums - missing file: %s", rel)
+			}
+			// it's okay if we're doing a partial check and the file isn't there as long as the path isn't in the list of paths to check
+			return nil
+		}
+
+		actualSHA, err := GetSHA256OfFile(path)
+		if err != nil {
+			return fmt.Errorf("unable to get checksum of: %s", err.Error())
+		}
+
+		if sha != actualSHA {
+			return fmt.Errorf("invalid checksum for %s: (expected: %s, received: %s)", path, sha, actualSHA)
+		}
+
+		pathCheckMap[path] = true
```

Comment:
```suggestion
		checkedMap[path] = true
```

CheckedMap may be more distinct from pathsToCheck (and implies better that they have already been checked)